### PR TITLE
Date slider refactor w/ fixed package-lock.json

### PR DESCRIPTION
**Pull request by atatmaja**
## Briefly describe the feature or bug that this PR addresses.
Refactors date slider to use a third-party component and condenses date slider.

## Please link the Airtable ticket associated with this PR.
[Condense date picker slider and calendar](https://airtable.com/appT8tnLbGJV6v81V/tbli2lWQHAqBa6ZcI/viwTYqtBDdt3Wt3Yo/rec4aoXDsOU7CmwAq?blocks=hide)

## If applicable, include screenshots of the feature/bugfix introduced by this PR (Please include both desktop and mobile if there is a significant UI change).
image

## Describe the steps you took to test the feature/bugfix introduced by this PR.
- Test updating date through input and seeing if slider adjusts & filter adjusts
- Test moving slider and see if filters adjust & input adjusts
- Test moving slider and seeing if input adjusts before mouseUp

## Does this PR depend on any recent backend work? If so, please include the PR number from the iit-backend repo and associated Airtable ticket link.
Nope!

## Does this PR require any French translations? Have they been included?
Nope!

### Explore

- Check Desktop and on Mobile
- Summary Stats
- Map loads in ~8 seconds with loading spinner 
- Map has pins
- Country Modal - check all fields
- Pins modal (check each grade) - check all fields
- Filter info bubbles
- Select each filter and check options
- Execute a Filter for Risk of bias, Sex, and Date.
- Check the "Database up to date to:" date
- Check each footer item (Privacy policy etc.)

### Analyse
- Check tableau embed loads and is functional
- Check for double scrolling (the embed scrolls within the page) 

### Data
- Click each button
- Click a FAQ for functionality
- References table embed loads
    - Check that downloads are disabled.
- Data download link provides access to a downloadable CSV.
    - Download the CSV
- "terms of use" button

### Publications
- Cards all load with images
- Carousel is clickable
- Try 2 random items on the page for functional links

### About
- Page loads
- Scroll to bottom

### Check serotracker.com/broken
- 404 page loads and animation works

### Check serotracker.com/Canada 
- Check tableau embed loads and is functional
- Check for double scrolling (the embed scrolls within the page) 
- Check in French

### CODE REVIEW

**Review by annoyingpr — APPROVED** [review]

File: src/components/sidebar/right-sidebar/Filters.tsx
Comment (atatmaja) [inline src/components/sidebar/right-sidebar/Filters.tsx:203]: Let's remove this comment

File: src/components/sidebar/right-sidebar/datepicker/Datepicker.tsx
Comment (atatmaja) [inline src/components/sidebar/right-sidebar/datepicker/Datepicker.tsx:13]: Do we need this import?

File: src/components/sidebar/right-sidebar/new-datepicker/NewDateSlider.tsx
Comment (atatmaja) [inline src/components/sidebar/right-sidebar/new-datepicker/NewDateSlider.tsx:13]: So minPossibleValue and maxPossibleValue are unix timestamps right? And values are offsets (in terms of milliseconds) from the min date?

Can we comment that explicitly in the code? It's a bit confusing to interpret otherwise

@annoyingpr @HimanshuRanka

File: src/components/sidebar/right-sidebar/new-datepicker/NewDateSlider.tsx
Comment (atatmaja) [inline src/components/sidebar/right-sidebar/new-datepicker/NewDateSlider.tsx:17]: Can we either add a comment describing onMouseUp's function signature or add some typing that will describe it better?

File: src/components/sidebar/right-sidebar/new-datepicker/NewDateSlider.tsx
Comment (atatmaja) [inline src/components/sidebar/right-sidebar/new-datepicker/NewDateSlider.tsx:1]: Also nit but let's just call the components `DatePicker` and `DateSlider`, they won't be new forever lmao

File: src/components/sidebar/right-sidebar/new-datepicker/NewDatepicker.tsx
Comment (atatmaja) [inline src/components/sidebar/right-sidebar/new-datepicker/NewDatepicker.tsx:42]: Reminder: will need to register de after the german PR is merged

File: src/components/sidebar/right-sidebar/new-datepicker/NewDatepicker.tsx
Comment (atatmaja) [inline src/components/sidebar/right-sidebar/new-datepicker/NewDatepicker.tsx:30]: probably

File: src/components/sidebar/right-sidebar/new-datepicker/NewDatepicker.tsx
Comment (atatmaja) [inline src/components/sidebar/right-sidebar/new-datepicker/NewDatepicker.tsx:79]: Need to translate

File: src/components/sidebar/right-sidebar/new-datepicker/NewDatepicker.tsx
Comment (atatmaja) [inline src/components/sidebar/right-sidebar/new-datepicker/NewDatepicker.tsx:111]: Is this text? If so we also need to translate

File: src/components/sidebar/right-sidebar/new-datepicker/components.tsx
Comment (atatmaja) [inline src/components/sidebar/right-sidebar/new-datepicker/components.tsx:37]: Probably easier to maintain in the long run if we just have each component in its own file here

File: src/components/sidebar/right-sidebar/new-datepicker/components.tsx
Comment (atatmaja) [inline src/components/sidebar/right-sidebar/new-datepicker/components.tsx:142]: can we use css classes?

File: src/components/sidebar/right-sidebar/new-datepicker/NewDateSlider.tsx
Comment (HimanshuRanka) [reply src/components/sidebar/right-sidebar/new-datepicker/NewDateSlider.tsx:13]: Yeah you got that right. I can add it in

File: src/components/sidebar/right-sidebar/new-datepicker/NewDatepicker.tsx
Comment (HimanshuRanka) [reply src/components/sidebar/right-sidebar/new-datepicker/NewDatepicker.tsx:42]: Maybe we should wait for German PR to get merged, and then fix it once and for all in this PR?

@annoyingpr @atatmaja

File: src/components/sidebar/right-sidebar/new-datepicker/NewDatepicker.tsx
Comment (HimanshuRanka) [reply src/components/sidebar/right-sidebar/new-datepicker/NewDatepicker.tsx:30]: Hmmm in hindsight I also feel it is fine here since they won't really be used outside of the datepicker files. Feel like moving it to util might just make it harder to reference when being reviewed in the future. thoughts?

File: src/components/sidebar/right-sidebar/new-datepicker/components.tsx
Comment (HimanshuRanka) [reply src/components/sidebar/right-sidebar/new-datepicker/components.tsx:37]: Might be unnecessary because all these components are from the same package. Just out of curiosity why would having an indiv file make it easier to maintain? (In the docx they have all comps in the same file - not that this is convention - but if we had to work on it together, would be easier to read the code if it was similar to the documentation)

Comment (atatmaja) [reply src/components/sidebar/right-sidebar/new-datepicker/components.tsx:37]: Consider the case where we know that the rail component specifically is buggy. Currently someone who wasn't super familiar with the code would have a difficult time figuring out where to start looking, since the component they would need to edit would be in a vaguely named `components.tsx` file. Discoverability would be better if it were in its own Rail.tsx file.

That being said, you guys are more familiar with the conventions here, so I'll leave the final judgement on what to do here up to you. Just a suggestion

File: src/components/sidebar/right-sidebar/datepicker/Datepicker.tsx
Comment (annoyingpr) [reply src/components/sidebar/right-sidebar/datepicker/Datepicker.tsx:13]: nope, removed!

File: src/components/sidebar/right-sidebar/new-datepicker/NewDateSlider.tsx
Comment (annoyingpr) [reply src/components/sidebar/right-sidebar/new-datepicker/NewDateSlider.tsx:1]: yeah good point LOL, ill remove the old ones and update the names

File: src/components/sidebar/right-sidebar/new-datepicker/NewDatepicker.tsx
Comment (annoyingpr) [reply src/components/sidebar/right-sidebar/new-datepicker/NewDatepicker.tsx:30]: i agree with @HimanshuRanka here!

File: src/components/sidebar/right-sidebar/new-datepicker/NewDatepicker.tsx
Comment (annoyingpr) [reply src/components/sidebar/right-sidebar/new-datepicker/NewDatepicker.tsx:42]: updated!

File: src/components/sidebar/right-sidebar/new-datepicker/components.tsx
Comment (annoyingpr) [reply src/components/sidebar/right-sidebar/new-datepicker/components.tsx:37]: I agree with @atatmaja here, split up each component into its own file for improved maintainability in the future!

File: src/components/sidebar/right-sidebar/new-datepicker/components.tsx
Comment (annoyingpr) [reply src/components/sidebar/right-sidebar/new-datepicker/components.tsx]: yup, added CSS classes to all the slider components :D

File: src/components/sidebar/right-sidebar/datepicker/Handle.tsx
Comment (atatmaja) [inline src/components/sidebar/right-sidebar/datepicker/Handle.tsx:16]: Nit: To be consistent with how we style elsewhere, we define these styles in .css files. If I'm not mistaken, divs can have both a className and style property, so we can do something like:

```
<div
  className="activeHandleStyle"
  style={{
    left: `${percent}%`,
  }}
>
```

**Review by atatmaja — COMMENTED** [review]
LGTM apart from nit - can't approve this PR cause I'm technically the author lmao

File: src/components/sidebar/right-sidebar/datepicker/DateSlider.css
Comment (HimanshuRanka) [inline src/components/sidebar/right-sidebar/datepicker/DateSlider.css:105]: Pretty sure bootstrap already has a utility class for space-between. so might not need this

File: src/components/sidebar/right-sidebar/datepicker/Handle.tsx
Comment (HimanshuRanka) [reply src/components/sidebar/right-sidebar/datepicker/Handle.tsx:16]: +1

File: src/utils/translate/fr.json
Comment (HimanshuRanka) [inline src/utils/translate/fr.json:456]: Do we have this added in the German json as well?

**Review by HimanshuRanka — CHANGES_REQUESTED** [review]

File: src/utils/translate/fr.json
Comment (annoyingpr) [reply src/utils/translate/fr.json:456]: yup, just used Google Translate for now

File: src/components/sidebar/right-sidebar/datepicker/DateSlider.css
Comment (annoyingpr) [reply src/components/sidebar/right-sidebar/datepicker/DateSlider.css:105]: seems wonky when i remove it, not sure if the utility class does what we need so ill just stick with what we have for now 

image

File: src/components/sidebar/right-sidebar/new-datepicker/components.tsx
Comment (HimanshuRanka) [reply src/components/sidebar/right-sidebar/new-datepicker/components.tsx]: Ahhhh yes that totally makes sense!

File: src/components/sidebar/right-sidebar/datepicker/Datepicker.tsx
Comment (annoyingpr) [inline src/components/sidebar/right-sidebar/datepicker/Datepicker.tsx:77]:
```suggestion
   
```

**Review by HimanshuRanka — APPROVED** [review]

**Pull request merged**
